File: Java/src/main/java/Solution401_500/Solution434.java
package Solution401_500;

/**
 * @author windows
 */
public class Solution434 {
    public int countSegments(String s) {
        s = s.trim();
        if (s.length() == 0) return 0;
        String[] strings = s.split("[ ]+");
        return strings.length;
    }



    public static void main(String[] args) {
        Solution434 solution434 = new Solution434();
    }
}

File: C++/src/Solution401_500/Solution434.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution434.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution434::countSegments(string s) {
    int result = 0;
    for (int i = 0; i < s.length(); ++i) {
        if ((i == 0 || s[i- 1] == ' ') && s[i] != ' '){
            result++;
        }
    }
    return result;
}
